### ISSUE TITLE
🐛[BUG]: `ewald_reciprocal_space` autograd silently returns a wrong virial

### ISSUE THREAD

**Issue by sklenard** [NVIDIA/nvalchemi-toolkit-ops#136]
### Version

0.4.0

### Describe the issue

Taking a strain/cell gradient through `ewald_reciprocal_space` produces a wrong virial, with no error and no warning. `compute_virial=True` returns the correct tensor, but it is deprecated.

### Minimum reproducible example

```python
import warnings

import torch
from nvalchemiops.torch.interactions.electrostatics import (
    ewald_reciprocal_space,
    generate_k_vectors_ewald_summation,
)

torch.set_default_dtype(torch.float64)

# 4 charges in a triclinic box
CELL = torch.tensor([[6.0, 0.0, 0.0], [1.0, 5.0, 0.0], [0.5, 0.7, 5.5]])
POS = torch.tensor([[0.5, 0.5, 0.5], [3.0, 1.0, 2.0], [1.5, 3.5, 4.0], [4.5, 2.5, 1.0]])
CHARGES = torch.tensor([1.0, -1.0, 0.7, -0.7])
ALPHA = torch.tensor([0.4])
K_CUTOFF = torch.tensor([4.0])

# built once from the unstrained cell: a constant w.r.t. the strain
K_PRECOMPUTED = generate_k_vectors_ewald_summation(cell=CELL, k_cutoff=K_CUTOFF)


def energy(pos, cell, k_vectors=None):
    """`k_vectors=None` rebuilds them from the strained `cell` (requires_grad)."""
    if k_vectors is None:
        k_vectors = generate_k_vectors_ewald_summation(cell=cell, k_cutoff=K_CUTOFF)
    return ewald_reciprocal_space(positions=pos, charges=CHARGES, cell=cell,
                                  k_vectors=k_vectors, alpha=ALPHA).sum()


def strain(eps):
    """Affine deformation of positions and cell by a symmetric strain."""
    sym = 0.5 * (eps + eps.T)
    return POS + POS @ sym, CELL + CELL @ sym


def virial_finite_difference(h=1e-6):
    """Reference, with the k-vectors following the strained cell."""
    dE = torch.zeros(3, 3)
    for a in range(3):
        for b in range(3):
            values = []
            for s in (+h, -h):
                eps = torch.zeros(3, 3)
                eps[a, b] += 0.5 * s
                eps[b, a] += 0.5 * s
                with torch.no_grad():
                    values.append(energy(*strain(eps)))
            dE[a, b] = (values[0] - values[1]) / (2 * h)
    return -dE  # nvalchemiops convention: W = -dE/d(strain)


def virial_autograd(k_vectors=None):
    eps = torch.zeros(3, 3, requires_grad=True)
    return -torch.autograd.grad(energy(*strain(eps), k_vectors), eps)[0]


def virial_kernel():
    with warnings.catch_warnings():  # the flag is deprecated
        warnings.simplefilter("ignore", DeprecationWarning)
        out = ewald_reciprocal_space(
            positions=POS, charges=CHARGES, cell=CELL.unsqueeze(0),
            k_vectors=K_PRECOMPUTED, alpha=ALPHA,
            batch_idx=torch.zeros(POS.shape[0], dtype=torch.int32),
            compute_virial=True,
        )
    return out[-1].reshape(3, 3)


def show(label, W, reference=None):
    verdict = ""
    if reference is not None:
        error = (W - reference).abs().max()
        verdict = f"max error = {error:.1e}  {'ok' if error < 1e-8 else 'WRONG'}"
    print(f"\n{label:<44}{verdict}")
    for row in W:
        print("   " + "  ".join(f"{v:+12.6f}" for v in row))


reference = virial_finite_difference()

show("finite difference (reference)", reference)
show("autograd, pre-computed k-vectors", virial_autograd(K_PRECOMPUTED), reference)
show("autograd, k-vectors from strained cell", virial_autograd(), reference)
show("compute_virial=True (deprecated)", virial_kernel(), reference)
```

### Relevant log output

```shell
finite difference (reference)               
      -0.419709     +0.108878     +0.022028
      +0.108878     +0.062030     -0.020228
      +0.022028     -0.020228     -0.061639

autograd, pre-computed k-vectors            max error = 5.8e-01  WRONG
      +0.165091     +0.026475     -0.049095
      +0.026475     +0.118494     -0.053209
      -0.049095     -0.053209     +0.087324

autograd, k-vectors from strained cell      max error = 5.8e-01  WRONG
      +0.165091     +0.026475     -0.049095
      +0.026475     +0.118494     -0.053209
      -0.049095     -0.053209     +0.087324

compute_virial=True (deprecated)            max error = 7.6e-11  ok
      -0.419709     +0.108878     +0.022028
      +0.108878     +0.062030     -0.020228
      +0.022028     -0.020228     -0.061639
```

### Environment details

```shell

```
